lib/browserLauncher.js

```diff
@@ -0,0 +1,370 @@
+
+    // Do not automatically fill sign-in forms with known usernames and
+    // passwords
+    'signon.autofillForms': false,
+    // Disable password capture, so that tests that include forms are not
+    // influenced by the presence of the persistent doorhanger notification
+    'signon.rememberSignons': false,
+
+    // Disable first-run welcome page
+    'startup.homepage_welcome_url': 'about:blank',
+
+    // Disable first-run welcome page
+    'startup.homepage_welcome_url.additional': '',
+
+    // Disable browser animations (tabs, fullscreen, sliding alerts)
+    'toolkit.cosmeticAnimations.enabled': false,
+
+    // We want to collect telemetry, but we don't want to send in the results
+    'toolkit.telemetry.server': `https://${server}/dummy/telemetry/`,
+    // Prevent starting into safe mode after application crashes
+    'toolkit.startup.max_resumed_crashes': -1,
+  };
+
+  Object.assign(defaultPreferences, extraPrefs);
+  for (const [key, value] of Object.entries(defaultPreferences)) {
+    userJS.push(`user_pref(${JSON.stringify(key)}, ${JSON.stringify(value)});`);
+  }
+  await writeFileAsync(path.join(profilePath, 'user.js'), userJS.join('\n'));
+  await writeFileAsync(path.join(profilePath, 'prefs.js'), prefsJS.join('\n'));
+  return profilePath;
+}
+
+async function setFirefoxBrowserArgs(options) {
+  let args = ['--no-remote', '--foreground', 'about:blank', '--remote-debugging-port=0'];
+  args = updateArgsFromOptions(args, options);
+  setHeadlessArgs(args, options);
+  if (!args.includes('-profile') && !args.includes('--profile')) {
+    temporaryUserDataDir = await createProfile(options.extraPrefsFirefox);
+    args.push('--profile');
+    args.push(temporaryUserDataDir);
+  }
+  return args;
+}
+
+async function setChromeBrowserArgs(options) {
+  let args = [
+    `--remote-debugging-port=${options.port}`,
+    '--disable-features=site-per-process,TranslateUI',
+    '--enable-features=NetworkService,NetworkServiceInProcess',
+    '--disable-renderer-backgrounding',
+    '--disable-backgrounding-occluded-windows',
+    '--disable-background-timer-throttling',
+    '--disable-background-networking',
+    '--disable-breakpad',
+    '--disable-default-apps',
+    '--disable-hang-monitor',
+    '--disable-prompt-on-repost',
+    '--disable-sync',
+    '--force-color-profile=srgb',
+    '--safebrowsing-disable-auto-update',
+    '--password-store=basic',
+    '--use-mock-keychain',
+    '--enable-automation',
+    '--disable-notifications',
+    'about:blank',
+  ];
+  args = updateArgsFromOptions(args, options);
+  if (!args.some((arg) => arg.startsWith('--user-data-dir'))) {
+    const os = require('os');
+    const CHROME_PROFILE_PATH = path.join(os.tmpdir(), 'taiko_dev_profile-');
+    temporaryUserDataDir = await mkdtempAsync(CHROME_PROFILE_PATH);
+    args.push(`--user-data-dir=${temporaryUserDataDir}`);
+  }
+  setHeadlessArgs(args, options);
+  return args;
+}
+
+function errorMessageForChromeProcessCrash() {
```

Comment:
Is there something similar for to handle firefox crash?

Reply:
This method will handle any browser process crash, updated the name accordingly.